lib/ansible/cli/inventory.py

```diff
@@ -152,7 +152,7 @@ def run(self):
                 try:
                     with open(to_bytes(outfile), 'wb') as f:
                         f.write(to_bytes(results))
-                except (OSError, IOError) as e:
+                except OSError as e:
                     raise AnsibleError('Unable to write to destination file (%s): %s' % (to_native(outfile), to_native(e)))
```

Comment:
When raising a new exception, be sure to use `from ex` to assign the cause, or `from None` if the cause is not useful to include in the traceback:

```suggestion
                    raise AnsibleError(f'Unable to write to destination file {outfile!r}.') from ex
```